src/Dispatcher.js

```diff
@@ -1,31 +1,56 @@
-import { PropTypes } from 'react';
-import createDispatcher from './createDispatcher';
+function dispatch(store, atom, action) {
+  return store(atom, action);
+}
 
-export default class ReduxDispatcher {
-  static propTypes = {
-    stores: PropTypes.object.isRequired,
-    children: PropTypes.func.isRequired
-  };
+export default class Dispatcher {
+  constructor(store) {
+    this.store = store;
+    this.hydrate();
+  }
 
-  static childContextTypes = {
-    redux: PropTypes.object.isRequired
-  };
+  hydrate({ atom, subscriptions = [] } = {}) {
+    this.atom = atom;
+    this.subscriptions = subscriptions;
+    this.dispatch({});
+  }
 
-  constructor(props) {
-    this.dispatcher = createDispatcher();
-    this.dispatcher.receiveStores(props.stores);
+  dehydrate() {
+    const { atom, subscriptions } = this;
+    delete this.atom;
+    this.subscriptions = [];
+    return { atom, subscriptions };
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.dispatcher.receiveStores(nextProps.stores);
+  dispatch(action) {
+    const nextAtom = dispatch(this.store, this.atom, action);
+    this.setAtom(nextAtom);
   }
 
-  getChildContext() {
-    return { redux: this.dispatcher };
+  perform(actionCreator, ...args) {
+    const action = actionCreator(...args);
+
+    return typeof action === 'function'
+      ? action(this.dispatch, this.atom)
+      : this.dispatch(action);
+  }
+
+  setAtom(atom) {
+    this.atom = atom;
+    this.emitChange();
+  }
+
+  subscribe(listener) {
+    this.subscriptions.push(listener);
+    listener(this.atom);
+
+    return () => {
+      const index = this.subscriptions.indexOf(listener);
+      this.subscriptions.splice(index, 1);
+    };
   }
 
-  render() {
-    const { children } = this.props;
-    return children();
+  emitChange() {
```

Comment:
There are also other problems with using just `setState`. For example, the state passed to action creator might be stale because there are pending operation that have not yet been rendered. We are able to sidestep this problem completely by maintaining the atom manually.